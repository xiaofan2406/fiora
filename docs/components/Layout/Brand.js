/* @flow */
import React from 'react';
import { css } from 'react-emotion';
import { fontSizes, spacing } from 'styles';
import { APP_TITLE } from 'utils/constants';
import logo from 'assets/logo.svg';

const cssBrand = css`
  display: flex;
  align-items: center;
  padding: ${spacing.internal}px;
  & > .logo {
    animation: spin infinite 10s linear;
    height: 36px;
  }
  & > .title {
    animation: fadeIn 2s ease;
    font-size: ${fontSizes.large}px;
  }
`;

class Brand extends React.Component<{}> {
  shouldComponentUpdate() {
    return false;
  }

  render() {
    return (
      <div className={cssBrand}>
        <img src={logo} alt="logo" className="logo" />
        <span className="title">{APP_TITLE}</span>
      </div>
    );
  }
}

export { Brand as Component };

export default Brand;
