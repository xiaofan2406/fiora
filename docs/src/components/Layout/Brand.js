import React from 'react';
import { css } from 'emotion';
import { variables, fontSizes, theme } from 'styles';
import { APP_TITLE } from 'configs';
import logo from 'assets/logo.svg';

const brand = css`
  display: flex;
  align-items: center;
  & img {
    animation: spin infinite 10s linear;
    height: ${variables.Brand.logoHeight}px;
  }
  & span {
    animation: fadeIn 2s ease;
    font-size: ${fontSizes.large}px;
    color: ${theme.inverseColor};
  }
`;

class Brand extends React.Component {
  shouldComponentUpdate() {
    return false;
  }

  render() {
    return (
      <div className={brand}>
        <img src={logo} alt="logo" />
        <span>{APP_TITLE}</span>
      </div>
    );
  }
}

export { Brand as Component };

export default Brand;
