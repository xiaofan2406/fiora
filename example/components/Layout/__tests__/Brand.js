import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Component } from '../Brand';

it('should match snapshot', () => {
  const props = { classes: {} };
  const component = shallow(<Component {...props} />);

  expect(toJson(component)).toMatchSnapshot();
});

it('should not update on props changes', () => {
  const renderSpy = jest.spyOn(Component.prototype, 'render');
  const props = { classes: {} };
  const component = shallow(<Component {...props} />);

  expect(renderSpy).toHaveBeenCalledTimes(1);

  component.setProps({
    classes: { someClass: {} }
  });

  expect(renderSpy).toHaveBeenCalledTimes(1);
});
