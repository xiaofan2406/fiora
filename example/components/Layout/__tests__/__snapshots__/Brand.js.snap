// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="css-1hph391"
>
  <img
    alt="logo"
    src="logo.svg"
  />
  <span>
    Fiora
  </span>
</div>
`;
